src/snowflake/connector/result_set.py

```diff
@@ -238,6 +264,9 @@ def _create_iter(
         This function is a helper function to ``__iter__`` and it was introduced for the
         cases where we need to propagate some values to later ``_download`` calls.
         """
+        # pop is_fetch_all and pass it to result_set_iterator
+        is_fetch_all = kwargs.pop("is_fetch_all", None)
```

Comment:
nit: default value `False`?
```suggestion
        is_fetch_all = kwargs.pop("is_fetch_all", False)
```